src/pid_ros.cpp

```diff
@@ -0,0 +1,328 @@
+/*********************************************************************
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2020, Open Source Robotics Foundation, Inc.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of the Open Source Robotics Foundation nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+#include <algorithm>
+#include <cmath>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "control_toolbox/pid_ros.hpp"
+
+namespace control_toolbox
+{
+
+void PidROS::initialize(std::string topic_prefix)
+{
+  if (topic_prefix.back() != '.' && !topic_prefix.empty()) {
+    topic_prefix_ = topic_prefix + ".";
+  } else {
+    topic_prefix_ = topic_prefix;
+  }
+
+  state_pub_ = rclcpp::create_publisher<control_msgs::msg::PidState>(
+    topics_interface_,
+    topic_prefix + "/pid_state",
+    rclcpp::SensorDataQoS());
+  rt_state_pub_.reset(
+    new realtime_tools::RealtimePublisher<control_msgs::msg::PidState>(state_pub_));
+}
+
+bool
+PidROS::getBooleanParam(const std::string & param_name, bool & value)
```

Comment:
This and `getDoubleParam` are basically implementing the same thing but I see that you needed more prints on the double version so that one is more evolved. Should we mark these already as to-be-removed once this functionality becomes supported by the param API directly?